services/viva-ms/lambdas/checkCompletion.js

```diff
@@ -2,14 +2,18 @@
 import to from 'await-to-js';
 
 import config from '../../../config';
-import params from '../../../libs/params';
+
 import * as dynamoDb from '../../../libs/dynamoDb';
-import validateApplicationStatus from '../helpers/validateApplicationStatus';
-import { getStatusByType } from '../../../libs/caseStatuses';
-import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import params from '../../../libs/params';
 import log from '../../../libs/logs';
+import { getStatusByType, statusTypes } from '../../../libs/caseStatuses';
+
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import validateApplicationStatus from '../helpers/validateApplicationStatus';
 
 const VIVA_CASE_SSM_PARAMS = params.read(config.cases.providers.viva.envsKeyName);
+const VIVA_COMPLETION_REQUIRED = 'VIVA_COMPLETION_REQUIRED';
 
 export async function main(event, context) {
   const { caseKeys } = event.detail;
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if (noSuchProperty('caseKeys', event.detail)){
  // do something
}

const {caseKeys} = event.detail
 if(noSuchProperty('PK', caseKeys) || noSuchProperty('SK', caseKeys)) {
   // do something
 }

```

Reply:
Will be fixed in future task